docs/pages/access-controls/access-graph/self-hosted-helm.mdx

```diff
@@ -0,0 +1,220 @@
+You can skip the final command if you are instead using cert-manager to issue the certificate,
+or otherwise have already provisioned the TLS certificate for Teleport Access Graph in your Kubernetes cluster.
+
+```code
+$ kubectl create namespace teleport-access-graph 
+$ kubectl -n teleport-access-graph create secret generic teleport-access-graph-postgres \
+    --from-literal uri="postgres://access_graph_user:my_password@db.example.com:5432/access_graph_db?sslmode=require"
+$ kubectl -n teleport-access-graph create secret tls teleport-access-graph-tls --cert=./certs/cert.crt --key=./certs/cert.key
+```
+
+The above commands assume that you have the TLS certificate and private key for the TAG server stored under the `./certs` directory.
+
+<Details title="Generating TLS certificates for development">
+The following script can be used to quickly generate a certificate authority and server TLS certificate for deploying TAG.
+
+```sh
+#!/usr/bin/env bash
+
+set -e
+
+mkdir -p ./certs
+cd ./certs
+
+openssl genrsa -aes256 -out ca.key 4096
+openssl req -x509 -new -key ca.key -sha256 -days 3652 -out ca.crt -subj '/CN=root'
+openssl req -new -out cert.csr -newkey rsa:4096 -nodes -keyout cert.key -subj '/CN=Access Graph'
+openssl x509 -req -in cert.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out cert.crt -days 3652 -sha256 \
+  -extfile <(printf 'extendedKeyUsage = serverAuth, clientAuth\nsubjectAltName = DNS:teleport-access-graph.teleport-access-graph.svc.cluster.local')
+
+echo "Certificates issued successfully"
+```
+
+The script will ask to input the same password 4 times:
+twice to create and confirm the password to encrypt the CA private key material with,
+then once to use the private key to self-sign the CA certificate,
+and finally once to use for signing the TAG server certificate.
+
+Please note that this script is only provided for proof-of-concept deployments,
+and may not follow the best security practices.
+</Details>
+
+After provisioning the secrets, create the following file named `tag-values.yaml` file for the TAG deployment,
+replacing the value under the `clusterHostCAs` key with the Teleport Host CA certificate retrieved previously.
+
+```yaml
+postgres:
+  secretName: "teleport-access-graph-postgres"
+
+tls:
+  # You may need to change this value if using cert-manager or other means
+  # to provision TLS certificates for your Kubernetes workloads.
+  existingSecretName: "teleport-access-graph-tls"
+
+clusterHostCAs:
+  # Replace this with the certificate retrieved in Step 1.
+  # `clusterHostCAs` is an array of multiline YAML strings, `- |` must not be removed, and the indentation must be kept.
+  - |
+    -----BEGIN CERTIFICATE-----
+    MIIDqjCCApKgAwIBAgIQMIK8/WiQ/rUOrjlmB0IHVTANBgkqhkiG9w0BAQsFADBv
+    <...>
+    -----END CERTIFICATE-----
+```
+
+Finally, deploy the TAG service using Helm:
+
+```code
+$ helm install -n teleport-access-graph -f tag-values.yaml teleport-access-graph teleport/teleport-access-graph
```

Comment:
You will want to pin the chart version here, or else people will get TAG upgrades by mistake, which can be incompatible with their Teleport version or bring breaking changes.